tests/test_generic.py

```diff
@@ -110,11 +111,21 @@
 )
 def test_generic(tin, tout):
     link = Link("http://localhost", content=get_sample("generic/%s" % tin))
-    preview = Generic(link, parser="html.parser")
+    preview = Generic(link, "html.parser", None)
     for key in tout.keys():
         assert getattr(preview, key) == tout[key]
 
 
+def test_bs4_object():
```

Comment:
@biolds Please don't touch other test cases, keep this and add another case for when both `parser` and `soup` are set.